proofs/eo/cpc/programs/Nfa.eo

```diff
@@ -0,0 +1,161 @@
+TODO - How could we use the new list operators in ethos? in particular eo::list_setof
+; declare types - probably some are redundant
+(define nfa.char () String)
+
+; could we use the types in re module? (e.g. re.range / re.allchar?)
+(declare-const nfa.epsilon nfa.char)
+(declare-const nfa.allchar nfa.char)
+(declare-const nfa.range (-> Int Int nfa.char))
+(declare-const nfa.push nfa.char)
+(declare-const nfa.pop nfa.char)
+(declare-const nfa.none nfa.char)
+
+(declare-type nfa.trans ())
+(declare-const nfa.trans.null nfa.trans)
+(declare-const nfa.trans.decline nfa.trans)
+(declare-const nfa.trans.list (-> nfa.trans nfa.trans nfa.trans) :right-assoc-nil nfa.trans.null)
+(declare-const nfa.trans.stack (-> nfa.trans nfa.trans nfa.trans) :right-assoc-nil nfa.trans.null)
+(declare-const nfa.trans.init (-> nfa.char nfa.trans nfa.trans))
+
+; make sure we in a list form
```

Comment:
Try to have a relatively formal documentation, in this style:
https://github.com/cvc5/cvc5/blob/main/proofs/eo/cpc/programs/Strings.eo

For each function.